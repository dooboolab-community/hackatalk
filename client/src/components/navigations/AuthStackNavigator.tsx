import React, {ReactElement} from 'react';
import {
  StackNavigationProp,
  createStackNavigator,
} from '@react-navigation/stack';

import {CompositeNavigationProp} from '@react-navigation/native';
import FindPwContainer from '../pages/FindPwContainer';
import LicenseAgreementContainer from '../pages/LicenseAgreementContainer';
import {RootStackNavigationProps} from './RootStackNavigator';
import SignInContainer from '../pages/SignInContainer';
import SignUp from '../pages/SignUp';
import VerifyEmail from '../pages/VerifyEmail';
import {getString} from '../../../STRINGS';
import {useTheme} from 'dooboo-ui';

export type AuthStackParamList = {
  default: undefined;
  SignIn: undefined;
  SignUp: undefined;
  FindPw: undefined;
  VerifyEmail: {
    email: string;
  };
  LicenseAgreement: undefined;
};

type NavigationProps<
  T extends keyof AuthStackParamList = 'default'
> = StackNavigationProp<AuthStackParamList, T>;

export type AuthStackNavigationProps<
  T extends keyof AuthStackParamList = 'default'
> = CompositeNavigationProp<
  NavigationProps<T>,
  RootStackNavigationProps<'AuthStack'>
>;

const Stack = createStackNavigator<AuthStackParamList>();

function AuthNavigator(): ReactElement {
  const {theme} = useTheme();

  return (
    <Stack.Navigator
      initialRouteName="SignIn"
      screenOptions={{
        headerStyle: {
          backgroundColor: theme.background,
          borderBottomWidth: 0,
          elevation: 0,
        },
        headerTintColor: theme.text,
      }}>
      <Stack.Screen
        name="SignIn"
        component={SignInContainer}
        options={{
          headerShown: false,
        }}
      />
      <Stack.Screen
        name="SignUp"
        component={SignUp}
        options={{
          title: getString('SIGN_UP'),
        }}
      />
      <Stack.Screen
        name="FindPw"
        component={FindPwContainer}
        options={{
          title: getString('FIND_PW'),
        }}
      />
      <Stack.Screen
        name="VerifyEmail"
        component={VerifyEmail}
        options={{
          title: getString('VERIFY_EMAIL'),
        }}
      />
      <Stack.Screen
        name="LicenseAgreement"
        component={LicenseAgreementContainer}
        options={{
          title: getString('LICENSE_AGREEMENT'),
        }}
      />
    </Stack.Navigator>
  );
}

export default AuthNavigator;
