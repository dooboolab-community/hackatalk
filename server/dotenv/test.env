JWT_SECRET=dooboolab
DATABASE_URL="postgresql://test:test!@localhost:5432/postgres?schema=test"
