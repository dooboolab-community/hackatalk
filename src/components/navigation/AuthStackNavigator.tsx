import React, { ReactElement, useEffect } from 'react';
import { StackNavigationProp, createStackNavigator } from '@react-navigation/stack';

import { CompositeNavigationProp } from '@react-navigation/native';
import FindPw from '../screen/FindPw';
import { RootStackNavigationProps } from '../navigation/RootStackNavigator';
import SignIn from '../screen/SignIn';
import SignUp from '../screen/SignUp';
import VerifyEmail from '../screen/VerifyEmail';
import { getString } from '../../../STRINGS';
import { useThemeContext } from '@dooboo-ui/native-theme';

export type AuthStackParamList = {
  default: undefined;
  SignIn: undefined;
  SignUp: undefined;
  FindPw: undefined;
  VerifyEmail: {
    email: string;
  };
};

type NavigationProps<
  T extends keyof AuthStackParamList = 'default'
> = StackNavigationProp<AuthStackParamList, T>;

export type AuthStackNavigationProps<
  T extends keyof AuthStackParamList = 'default'
> = CompositeNavigationProp<
NavigationProps<T>,
RootStackNavigationProps<'AuthStack'>
>;

const Stack = createStackNavigator<AuthStackParamList>();

function AuthNavigator(): ReactElement {
  const { theme } = useThemeContext();

  return (
    <Stack.Navigator
      initialRouteName="SignIn"
      screenOptions={{
        headerStyle: {
          backgroundColor: theme.background,
          borderBottomWidth: 0,
          elevation: 0,
        },
        headerTintColor: theme.fontColor,
      }}
    >
      <Stack.Screen name="SignIn" component={SignIn} options={{
        headerShown: false,
      }} />
      <Stack.Screen name="SignUp" component={SignUp} options={{
        title: getString('SIGN_UP'),
      }} />
      <Stack.Screen
        name="FindPw"
        component={FindPw}
        options={{
          title: getString('FIND_PW'),
        }}
      />
      <Stack.Screen
        name="VerifyEmail"
        component={VerifyEmail}
        options={{
          title: getString('VERIFY_EMAIL'),
        }}
      />
    </Stack.Navigator>
  );
}

export default AuthNavigator;
