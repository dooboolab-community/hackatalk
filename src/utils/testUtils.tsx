import React, { ReactElement } from 'react';

import RootProvider from '../providers';
import { ThemeType } from '../types';

export const createTestElement = (
  child: ReactElement,
  themeType?: ThemeType,
): ReactElement => (
  <RootProvider initialThemeType={themeType}>{child}</RootProvider>
);

export const createTestProps = (
  obj: object = {},
  moreScreenProps?: object,
): object | unknown | any => ({
  navigation: {
    navigate: jest.fn(),
    goBack: jest.fn(),
    resetRoot: jest.fn(),
  },
  screenProps: {
    changeThemeType: jest.fn(),
    ...moreScreenProps,
  },
  ...obj,
});
