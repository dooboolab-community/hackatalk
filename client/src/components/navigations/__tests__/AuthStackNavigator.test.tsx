import 'react-native';

import AuthStackNavigator from '../AuthStackNavigator';
import {NavigationContainer} from '@react-navigation/native';
import React from 'react';
import {createTestElement} from '../../../../test/testUtils';
import {render} from '@testing-library/react-native';

jest.mock('../../../components/pages/SignIn/SocialSignInButton', () => 'test');

describe('[Stack] navigator', () => {
  it('should renders without crashing', async () => {
    const component = createTestElement(
      <NavigationContainer>
        <AuthStackNavigator />
      </NavigationContainer>,
    );

    const screen = render(component);
    const json = screen.toJSON();

    expect(json).toBeTruthy();
    expect(json).toMatchSnapshot();
  });
});
