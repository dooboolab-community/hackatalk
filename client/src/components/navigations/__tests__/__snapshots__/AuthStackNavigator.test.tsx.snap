// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Stack] navigator should renders without crashing 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
              "flex": 1,
            },
            undefined,
          ]
        }
      >
        <RNSScreenContainer
          onLayout={[Function]}
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <RNSScreen
            activityState={2}
            forwardedRef={[Function]}
            pointerEvents="box-none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <View
              collapsable={false}
              style={
                Object {
                  "opacity": 1,
                }
              }
            />
            <View
              accessibilityElementsHidden={false}
              closing={false}
              gestureVelocityImpact={0.3}
              importantForAccessibility="auto"
              onClose={[Function]}
              onGestureBegin={[Function]}
              onGestureCanceled={[Function]}
              onGestureEnd={[Function]}
              onOpen={[Function]}
              onTransition={[Function]}
              pointerEvents="box-none"
              style={
                Array [
                  Object {
                    "display": "flex",
                    "overflow": undefined,
                  },
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                ]
              }
              transitionSpec={
                Object {
                  "close": Object {
                    "animation": "spring",
                    "config": Object {
                      "damping": 500,
                      "mass": 3,
                      "overshootClamping": true,
                      "restDisplacementThreshold": 10,
                      "restSpeedThreshold": 10,
                      "stiffness": 1000,
                    },
                  },
                  "open": Object {
                    "animation": "spring",
                    "config": Object {
                      "damping": 500,
                      "mass": 3,
                      "overshootClamping": true,
                      "restDisplacementThreshold": 10,
                      "restSpeedThreshold": 10,
                      "stiffness": 1000,
                    },
                  },
                }
              }
            >
              <View
                pointerEvents="box-none"
                style={
                  Object {
                    "flex": 1,
                    "marginTop": 0,
                  }
                }
              >
                <View
                  collapsable={false}
                  forwardedRef={[Function]}
                  needsOffscreenAlphaCompositing={false}
                  onGestureHandlerEvent={[Function]}
                  onGestureHandlerStateChange={[Function]}
                  style={
                    Object {
                      "flex": 1,
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                        Object {
                          "translateX": 0,
                        },
                      ],
                    }
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "flex": 1,
                          "overflow": "hidden",
                        },
                        Array [
                          Object {
                            "backgroundColor": "rgb(242, 242, 242)",
                          },
                          Object {
                            "backgroundColor": "#000",
                          },
                        ],
                      ]
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                          "flexDirection": "column-reverse",
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "flex": 1,
                          }
                        }
                      >
                        <RCTSafeAreaView
                          emulateUnlessSupported={true}
                          style={
                            Array [
                              Object {
                                "backgroundColor": "#000",
                                "flexBasis": 0,
                                "flexGrow": 1,
                                "flexShrink": 1,
                                "justifyContent": "center",
                              },
                              undefined,
                            ]
                          }
                        >
                          <RCTScrollView
                            style={
                              Array [
                                Object {
                                  "alignSelf": "center",
                                  "width": "100%",
                                },
                                undefined,
                              ]
                            }
                          >
                            <View>
                              <mocked-view
                                onPress={[Function]}
                                style={
                                  Object {
                                    "animatedStyle": Object {
                                      "current": Object {},
                                    },
                                    "initial": Object {
                                      "updater": [Function],
                                      "value": Object {
                                        "height": 160,
                                        "left": 525,
                                        "position": "absolute",
                                        "top": 400.2,
                                        "width": 160,
                                        "zIndex": 15,
                                      },
                                    },
                                    "viewDescriptors": Object {
                                      "add": [Function],
                                      "batchToRemove": Set {},
                                      "items": Array [],
                                      "rebuildsharableViewDescriptors": [Function],
                                      "remove": [Function],
                                      "sharableViewDescriptors": MutableValue {
                                        "_animation": null,
                                        "_id": 4,
                                        "_listeners": Array [],
                                        "_setter": [Function],
                                        "_value": Array [],
                                      },
                                      "tags": Set {},
                                      "waitForInsertSync": false,
                                      "waitForRemoveSync": false,
                                    },
                                    "viewsRef": Object {
                                      "add": [Function],
                                      "items": Set {},
                                      "remove": [Function],
                                    },
                                  }
                                }
                                testID="theme-test"
                              >
                                <Image
                                  source={
                                    Object {
                                      "canInstrument": true,
                                      "createTransformer": [Function],
                                      "getCacheKey": [Function],
                                      "process": [Function],
                                    }
                                  }
                                  style={
                                    Object {
                                      "height": "100%",
                                      "resizeMode": "cover",
                                      "width": "100%",
                                    }
                                  }
                                />
                              </mocked-view>
                              <View
                                style={
                                  Array [
                                    Object {
                                      "marginBottom": 40,
                                      "marginLeft": 40,
                                      "marginRight": 40,
                                      "marginTop": 40,
                                    },
                                    undefined,
                                  ]
                                }
                              >
                                <View
                                  style={
                                    Array [
                                      Object {
                                        "marginBottom": 72,
                                        "marginTop": 44,
                                      },
                                      undefined,
                                    ]
                                  }
                                >
                                  <View
                                    style={
                                      Object {
                                        "height": 72,
                                      }
                                    }
                                  />
                                  <Text
                                    style={
                                      Array [
                                        Object {
                                          "alignSelf": "flex-start",
                                          "color": "white",
                                          "fontSize": 20,
                                          "fontWeight": "bold",
                                          "marginLeft": 6,
                                        },
                                        undefined,
                                      ]
                                    }
                                  >
                                    Hello
                                  </Text>
                                </View>
                                <View
                                  style={
                                    Array [
                                      Object {
                                        "alignSelf": "stretch",
                                        "flexDirection": "column",
                                      },
                                      Object {
                                        "marginBottom": 4,
                                      },
                                    ]
                                  }
                                  testID="container-id"
                                >
                                  <View
                                    style={
                                      Array [
                                        Array [
                                          Object {
                                            "alignItems": "center",
                                            "alignSelf": "stretch",
                                            "borderBottomColor": "white",
                                            "borderBottomWidth": 1,
                                            "flexDirection": "row",
                                            "justifyContent": "space-between",
                                          },
                                          Object {
                                            "borderColor": "white",
                                          },
                                        ],
                                        false,
                                        Object {
                                          "borderColor": "white",
                                        },
                                        Object {
                                          "borderBottomColor": "white",
                                        },
                                      ]
                                    }
                                  >
                                    <Text
                                      style={
                                        Array [
                                          Array [
                                            Object {
                                              "color": "#6D6D6D",
                                              "fontSize": 14,
                                              "marginRight": 2,
                                            },
                                            Object {
                                              "width": 80,
                                            },
                                          ],
                                          false,
                                          Object {},
                                        ]
                                      }
                                    >
                                      Email
                                    </Text>
                                    <TextInput
                                      allowFontScaling={true}
                                      autoCapitalize="none"
                                      editable={true}
                                      multiline={false}
                                      onBlur={[Function]}
                                      onChangeText={[Function]}
                                      onFocus={[Function]}
                                      onSubmitEditing={[Function]}
                                      placeholder="hello@example.com"
                                      placeholderTextColor="#6D6D6D"
                                      rejectResponderTermination={true}
                                      secureTextEntry={false}
                                      style={
                                        Array [
                                          Array [
                                            Object {
                                              "color": "white",
                                              "flex": 1,
                                              "fontSize": 14,
                                              "paddingHorizontal": 8,
                                              "paddingVertical": 10,
                                            },
                                            Object {
                                              "color": "white",
                                              "height": 52,
                                            },
                                          ],
                                          false,
                                        ]
                                      }
                                      testID="input-email"
                                      underlineColorAndroid="transparent"
                                      value=""
                                    />
                                  </View>
                                  <View
                                    style={
                                      Object {
                                        "flex": 1,
                                        "flexDirection": "row",
                                        "justifyContent": "space-between",
                                      }
                                    }
                                  >
                                    <Text
                                      style={
                                        Array [
                                          Array [
                                            Object {
                                              "color": "#FF002E",
                                              "fontSize": 12,
                                              "marginTop": 4,
                                            },
                                            undefined,
                                          ],
                                          Object {
                                            "color": "#FF002E",
                                            "flex": 1,
                                          },
                                        ]
                                      }
                                    >
                                      
                                    </Text>
                                  </View>
                                </View>
                                <View
                                  style={
                                    Array [
                                      Object {
                                        "alignSelf": "stretch",
                                        "flexDirection": "column",
                                      },
                                      Object {
                                        "marginBottom": 20,
                                      },
                                    ]
                                  }
                                  testID="container-id"
                                >
                                  <View
                                    style={
                                      Array [
                                        Array [
                                          Object {
                                            "alignItems": "center",
                                            "alignSelf": "stretch",
                                            "borderBottomColor": "white",
                                            "borderBottomWidth": 1,
                                            "flexDirection": "row",
                                            "justifyContent": "space-between",
                                          },
                                          Object {
                                            "borderColor": "white",
                                          },
                                        ],
                                        false,
                                        Object {
                                          "borderColor": "white",
                                        },
                                        Object {
                                          "borderBottomColor": "white",
                                        },
                                      ]
                                    }
                                  >
                                    <Text
                                      style={
                                        Array [
                                          Array [
                                            Object {
                                              "color": "#6D6D6D",
                                              "fontSize": 14,
                                              "marginRight": 2,
                                            },
                                            Object {
                                              "width": 80,
                                            },
                                          ],
                                          false,
                                          Object {},
                                        ]
                                      }
                                    >
                                      Password
                                    </Text>
                                    <TextInput
                                      allowFontScaling={true}
                                      autoCapitalize="none"
                                      editable={true}
                                      multiline={false}
                                      onBlur={[Function]}
                                      onChangeText={[Function]}
                                      onFocus={[Function]}
                                      onSubmitEditing={[Function]}
                                      placeholder="******"
                                      placeholderTextColor="#6D6D6D"
                                      rejectResponderTermination={true}
                                      secureTextEntry={true}
                                      style={
                                        Array [
                                          Array [
                                            Object {
                                              "color": "white",
                                              "flex": 1,
                                              "fontSize": 14,
                                              "paddingHorizontal": 8,
                                              "paddingVertical": 10,
                                            },
                                            Object {
                                              "color": "white",
                                              "height": 52,
                                            },
                                          ],
                                          false,
                                        ]
                                      }
                                      testID="input-password"
                                      underlineColorAndroid="transparent"
                                      value=""
                                    />
                                  </View>
                                  <View
                                    style={
                                      Object {
                                        "flex": 1,
                                        "flexDirection": "row",
                                        "justifyContent": "space-between",
                                      }
                                    }
                                  >
                                    <Text
                                      style={
                                        Array [
                                          Array [
                                            Object {
                                              "color": "#FF002E",
                                              "fontSize": 12,
                                              "marginTop": 4,
                                            },
                                            undefined,
                                          ],
                                          Object {
                                            "color": "#FF002E",
                                            "flex": 1,
                                          },
                                        ]
                                      }
                                    >
                                      
                                    </Text>
                                  </View>
                                </View>
                                <View
                                  style={
                                    Array [
                                      Object {
                                        "flexDirection": "row",
                                        "height": 52,
                                        "marginTop": 4,
                                        "width": "100%",
                                      },
                                      undefined,
                                    ]
                                  }
                                >
                                  <View
                                    accessible={true}
                                    focusable={true}
                                    onClick={[Function]}
                                    onResponderGrant={[Function]}
                                    onResponderMove={[Function]}
                                    onResponderRelease={[Function]}
                                    onResponderTerminate={[Function]}
                                    onResponderTerminationRequest={[Function]}
                                    onStartShouldSetResponder={[Function]}
                                    style={
                                      Object {
                                        "flex": 1,
                                        "opacity": 1,
                                      }
                                    }
                                    testID="btn-sign-up"
                                  />
                                  <View
                                    style={
                                      Object {
                                        "width": 12,
                                      }
                                    }
                                  />
                                  <View
                                    accessible={true}
                                    focusable={true}
                                    onClick={[Function]}
                                    onResponderGrant={[Function]}
                                    onResponderMove={[Function]}
                                    onResponderRelease={[Function]}
                                    onResponderTerminate={[Function]}
                                    onResponderTerminationRequest={[Function]}
                                    onStartShouldSetResponder={[Function]}
                                    style={
                                      Object {
                                        "flex": 1,
                                        "opacity": 1,
                                      }
                                    }
                                    testID="btn-sign-in"
                                  />
                                </View>
                                <View
                                  accessible={true}
                                  focusable={true}
                                  onClick={[Function]}
                                  onResponderGrant={[Function]}
                                  onResponderMove={[Function]}
                                  onResponderRelease={[Function]}
                                  onResponderTerminate={[Function]}
                                  onResponderTerminationRequest={[Function]}
                                  onStartShouldSetResponder={[Function]}
                                  style={
                                    Object {
                                      "alignSelf": "center",
                                      "marginBottom": 28,
                                      "opacity": 1,
                                      "paddingBottom": 16,
                                      "paddingLeft": 16,
                                      "paddingRight": 16,
                                      "paddingTop": 16,
                                    }
                                  }
                                  testID="btn-find-pw"
                                >
                                  <Text
                                    style={
                                      Array [
                                        Object {
                                          "color": "#8BC9FF",
                                          "textDecorationLine": "underline",
                                        },
                                        undefined,
                                      ]
                                    }
                                  >
                                    Forgot password?
                                  </Text>
                                </View>
                                <View
                                  style={
                                    Array [
                                      Object {
                                        "marginBottom": 24,
                                      },
                                      undefined,
                                    ]
                                  }
                                >
                                  <View
                                    accessible={true}
                                    focusable={true}
                                    onClick={[Function]}
                                    onResponderGrant={[Function]}
                                    onResponderMove={[Function]}
                                    onResponderRelease={[Function]}
                                    onResponderTerminate={[Function]}
                                    onResponderTerminationRequest={[Function]}
                                    onStartShouldSetResponder={[Function]}
                                    style={
                                      Object {
                                        "marginBottom": 12,
                                        "opacity": 1,
                                      }
                                    }
                                    testID="btn-apple"
                                  />
                                  <test
                                    onUserCreated={[Function]}
                                    socialProvider="facebook"
                                    svgIcon={
                                      <SvgMock
                                        fill="#345997"
                                        height={18}
                                        width={18}
                                      />
                                    }
                                  />
                                  <test
                                    onUserCreated={[Function]}
                                    socialProvider="google"
                                    svgIcon={
                                      <SvgMock
                                        fill="#E04238"
                                        height={20}
                                        width={20}
                                      />
                                    }
                                  />
                                </View>
                                <View
                                  style={
                                    Array [
                                      Object {
                                        "flexDirection": "row",
                                        "flexWrap": "wrap",
                                        "justifyContent": "center",
                                        "paddingBottom": 40,
                                        "paddingLeft": 0,
                                        "paddingRight": 0,
                                        "paddingTop": 0,
                                      },
                                      undefined,
                                    ]
                                  }
                                >
                                  <Text
                                    style={
                                      Array [
                                        Object {
                                          "color": "white",
                                          "fontSize": 14,
                                        },
                                        Object {
                                          "textAlign": "center",
                                        },
                                      ]
                                    }
                                  >
                                    We consider that you agree with 
                                    <Text
                                      onPress={[Function]}
                                      style={
                                        Array [
                                          Object {
                                            "color": "white",
                                            "fontSize": 14,
                                          },
                                          Object {
                                            "textDecorationLine": "underline",
                                          },
                                        ]
                                      }
                                    >
                                      Terms for agreement
                                    </Text>
                                     and 
                                    <Text
                                      onPress={[Function]}
                                      style={
                                        Array [
                                          Object {
                                            "color": "white",
                                            "fontSize": 14,
                                          },
                                          Object {
                                            "textDecorationLine": "underline",
                                          },
                                        ]
                                      }
                                    >
                                      Privacy and Policy
                                    </Text>
                                     by going onto next step.
                                  </Text>
                                </View>
                              </View>
                            </View>
                          </RCTScrollView>
                        </RCTSafeAreaView>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </RNSScreen>
        </RNSScreenContainer>
        <View
          pointerEvents="box-none"
          style={
            Object {
              "height": 44,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "zIndex": 1,
            }
          }
        />
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
